use-cases.md

```diff
@@ -53,11 +53,7 @@ When a server wants to trace execution of a request, it generally needs to go th
 Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
> imo this makes the OT implementation work much harder than it needs to. 

It doesn't. Your existing approaches are still there.

    tracer.get_span_injector(map.class).inject(span, map)
    span = tracer.get_span_extractor(map.class).extract(map)

    tracer.get_span_injector(byte_buffer.class).inject(span, byte_buffer)
    span = tracer.get_span_extractor(byte_buffer.class).extract(byte_buffer)

> For the end user it looks like magic, and implementation needs to know about 100s of various rpc request types. 

No it doesn't. Because standard helper implementations are provided.

For the popular rpc request types we can provide implementations, but these too should be rather thin re-using the helper implementations [eg  tracer.get_span_extractor(map.class)]

> the classpath dependency of a tracer will become huge.

No it won't. These dependencies will of course be marked as optional or provided.

> In the current API, the OT implementor only needs to support two representations, a binary blob and a string key/value map. 

This won't support all rpc request types, it won't be optimised for all rpc request types, and it ties the API down to these limitations.